### PR TITLE
Migrate core to TS p2

### PR DESCRIPTION
**Pull request by albertogasparin**
Another batch of files from `loot-core` migrated to TS (loose mode)

### CODE REVIEW

File: packages/loot-core/src/server/spreadsheet/graph-data-structure.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ function Graph(serialized) {
     removeIncomingEdges(node);
     edges.delete(node);
     incomingEdges.delete(node);
+    return graph;
```
Comment (albertogasparin) [inline packages/loot-core/src/server/spreadsheet/graph-data-structure.js:44]: There was a test asserting removeNode returned self, but it was actually not 😕

File: packages/loot-core/src/server/spreadsheet/new/lexer.ts [hunk 2]
```diff
@@ -259,7 +266,7 @@ class Tokenizer {
     if (this.current() === '\n') {
       this.lineno--;
 
-      var idx = this.src.lastIndexOf('\n', this.index - 1);
+      let idx = this.str.lastIndexOf('\n', this.index - 1);
```
Comment (albertogasparin) [inline packages/loot-core/src/server/spreadsheet/new/lexer.ts:269]: Unless I got this wrong,`this.src` is not defined on the class so this line would blow up if hit. I've assumed it was a typo and we meant `this.str` instead.

File: packages/loot-core/src/server/spreadsheet/tests/graph.ts [hunk 3]
```diff
@@ -3,12 +3,12 @@ import assert from 'assert';
 
 // If using from the NPM package, this line would be
 // import Graph from "graph-data-structure";
-import Graph from '../data-compute/graph-data-structure';
+import Graph from '../graph-data-structure';
```
Comment (j-f1) [inline packages/loot-core/src/server/spreadsheet/tests/graph.ts:6]: This test file can be cleaned up entirely, I think none of the tests actually work and they are not run by the tester.

File: packages/loot-core/src/server/spreadsheet/new/nodes.ts [hunk 4]
```diff
@@ -1,14 +1,12 @@
 class Node {
-  constructor(lineno, colno, fieldNames, ...fields) {
+  colno;
```
Comment (albertogasparin) [inline packages/loot-core/src/server/spreadsheet/new/nodes.ts:2]: Had to refactor these classes more than expected as TS + prototypal inheritance = pain with dynamic fields. So we no longer assign those fields to Node but to the extended class (which might make more sense).

**Review by j-f1 — APPROVED** [review]
LGTM, thanks for continuing this work!

**Pull request merged**